public/app/features/profile/SignupForm.tsx

```diff
@@ -0,0 +1,135 @@
+      delete formData.name;
+    }
+
+    const response = await getBackendSrv().post('/api/user/signup/step2', {
+      email: formData.email,
+      code: formData.code,
+      username: formData.email,
+      orgName: formData.orgName,
+      password: formData.password,
+      name: formData.name,
+    });
+
+    if (response.code === 'redirect-to-select-org') {
+      window.location.href = getConfig().appSubUrl + '/profile/select-org?signup=1';
+    }
+    window.location.href = getConfig().appSubUrl + '/';
+  };
+
+  const routeParams = props.routeParams || {};
+  const defaultValues = {
+    orgName: routeParams.email,
+    email: routeParams.email,
+    username: routeParams.email,
+    code: routeParams.code,
+    name: routeParams.name,
+  };
+
+  return (
+    <Forms.Form defaultValues={defaultValues} onSubmit={onSubmit}>
+      {({ register, errors }) => {
+        return (
+          <>
+            {verifyEmailEnabled && (
+              <Forms.Field label="Email verification code (sent to your email)">
+                <Forms.Input name="code" size="md" ref={register} placeholder="Code" />
+              </Forms.Field>
+            )}
+            {!autoAssignOrg && (
+              <Forms.Field label="Org. name">
+                <Forms.Input size="md" name="orgName" placeholder="Org. name" ref={register} />
+              </Forms.Field>
+            )}
+            <Forms.Field label="Your name">
+              <Forms.Input size="md" name="name" placeholder="(optional)" ref={register} />
+            </Forms.Field>
+            <Forms.Field label="Email" invalid={!!errors.email} error={!!errors.email && errors.email.message}>
+              <Forms.Input
+                size="md"
+                name="email"
+                type="email"
+                placeholder="Email"
+                ref={register({
+                  required: 'Email is required',
+                  pattern: {
+                    value: /^\S+@\S+$/,
+                    message: 'Email is invalid',
+                  },
+                })}
+              />
+            </Forms.Field>
+            <Forms.Field
+              label="Password"
+              invalid={!!errors.password}
+              error={!!errors.password && errors.password.message}
+            >
+              <Forms.Input
+                size="md"
+                name="password"
+                type="password"
+                placeholder="Password"
+                ref={register({ required: 'Password is required' })}
+              />
+            </Forms.Field>
+
+            <Forms.Button type="submit">Submit</Forms.Button>
+            <span className={buttonSpacing}>
+              <Forms.LinkButton href={getConfig().appSubUrl + '/login'} variant="secondary">
+                Back
+              </Forms.LinkButton>
+            </span>
+          </>
+        );
+      }}
+    </Forms.Form>
+  );
+};
+
+const mapStateToProps = (state: StoreState) => ({
```

Comment:
Nit -> can you please move redux connection to the SignupPage and pass default data from there?